### PR TITLE
chore: upgrade zk to v1.0 api, refactor script and add ut

### PR DESCRIPTION
**Pull request by Y-Rookie**

### CODE REVIEW

**Review by kubeJocker — DISMISSED** [review]
LGTM

File: addons/zookeeper/templates/cmpd.yaml [hunk 1]
```diff
@@ -43,27 +44,63 @@ spec:
...
+      value: {{ .Values.metrics.enabled | quote }}
+    - name: ZOOKEEPER_METRICS_PORT
+      value: {{ .Values.metrics.port | quote }}
+    - name: ZOOKEEPER_DATA_DIR
+      value: {{ .Values.zookeeper.dataDir }}
+    - name: ZOOKEEPER_DATA_LOG_DIR
+      value: {{ .Values.zookeeper.dataLogDir }}
+    - name: ZOOBINDIR
+      value: "/opt/bitnami/zookeeper/bin"
+    - name: ZOOBIN
```
Comment (zjx20) [inline addons/zookeeper/templates/cmpd.yaml:64]: Is this var useful? Its value is the same as the upper one.

File: addons/zookeeper/templates/cmpd.yaml [hunk 2]
```diff
@@ -43,27 +44,63 @@ spec:
...
+      value: "/opt/bitnami/zookeeper/bin"
+    - name: ZOO_ENABLE_AUTH
+      value: "yes"
+    - name: SERVICE_PORT
```
Comment (zjx20) [inline addons/zookeeper/templates/cmpd.yaml:68]: This var seems useless.

Comment (kubeJocker) [reply addons/zookeeper/templates/cmpd.yaml:68]: service_port needs to be preserved as Gemini will utilize this variable.

Comment (zjx20) [reply addons/zookeeper/templates/cmpd.yaml:68]: please add a comment to the code.

File: addons/zookeeper/templates/cmpd.yaml (hunk 1 above)
Comment (Y-Rookie) [reply addons/zookeeper/templates/cmpd.yaml:64]: > Is this var useful? Its value is the same as the upper one.

It should be necessary. This addon uses the Bitnami image, and the Bitnami image internally has some dependent environment variables (env).

Comment (Y-Rookie) [reply addons/zookeeper/templates/cmpd.yaml:64]: > > Is this var useful? Its value is the same as the upper one.
> 
> It should be necessary. This addon uses the Bitnami image, and the Bitnami image internally has some dependent environment variables (env).

@kubeJocker please take a look

Comment (kubeJocker) [reply addons/zookeeper/templates/cmpd.yaml:64]: This var is used in the ZkServer.sh of version 3.4, so it is necessary.

Comment (zjx20) [reply addons/zookeeper/templates/cmpd.yaml:64]: Got it, please add a comment as well.

File: addons/zookeeper/templates/cmpd.yaml (hunk 2 above)
Comment (Y-Rookie) [reply addons/zookeeper/templates/cmpd.yaml:70]: > please add a comment to the code.

done

File: addons/zookeeper/templates/cmpd.yaml (hunk 1 above)
Comment (Y-Rookie) [reply addons/zookeeper/templates/cmpd.yaml:65]: > Got it, please add a comment as well.

done

**Review by zjx20 — APPROVED** [review]

**Review by wangyelei — APPROVED** [review]

**Pull request merged**